tests/cloudpickle_test.py

```diff
@@ -2130,6 +2130,53 @@ def test_pickle_importable_typevar(self):
         from typing import AnyStr
         assert AnyStr is pickle_depickle(AnyStr, protocol=self.protocol)
 
+    @unittest.skipIf(sys.version_info < (3, 7),
+                     "Pickling generics not supported below py37")
+    def test_generic(self):
```

Comment:
```suggestion
    def test_dynamic_class_inheriting_from_generic_type(self):
```

Reply:
I think it's good to have tests that check for the pickling of the (generic) type objects alone and then we can have "integration" test that checks for (dynamic) class definition annotated with such type hints.

Reply:
Why not, but then I would suggest to keep the two separate: have one test for the new feature this PR introduces (supporting the pickling of classes defined using `class A(Generic[T])` ), and have one test to test the pickling of constructs of the `typing` module.

By the way, I think we don't have the same definition of `Generic`. By `Generic`, I mean instances of the `typing._GenericAlias`, creating using `typing.Generic[...]`. It seems that you use "generic" for another definition, which I don't clearly understand. This is also why I think these test should not really be introduced by this PR.

Reply:
I just added such a test in 7fd4412. @valtron feel free to factorize a helper function that generates the list of all type definitions to reuse in both tests.

Reply:
BTW I forgot that I already added the `subprocess_worker` helper tool in the past :) We should use it more.